### ISSUE TITLE
[native filter] default value did not show when refresh page if user delete or select new value on dashboard

### ISSUE THREAD

**Issue by jinghua-qa** [apache/superset#16068]
When I set a default value for native filter, I remove the default value or change to a new value on dashboard native filter panel, then after refresh page the default value did not show
, it changed my most recent edit.

### Expected results

Default value will show when refresh page 

### Actual results

Default value will not show when refresh page if user have changed or deleted on dashboard native filter panel

#### Screenshots

https://user-images.githubusercontent.com/81597121/128232847-5422415b-bb76-415b-b4c8-ee4f4a0f553e.mov

#### How to reproduce the bug

1. Create a value native filter and select default value
2. go to dashboard and delete or change the default value, apply
3. Refresh page

### Environment

(please complete the following information):

- superset version: `latest master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x ] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by michael-s-molina** [comment]
@jinghua-qa This is not a bug. The default value is different from the current user selection. The default value is applied when a user does not have current selections applied. When a user selects and applies a value, his selection is stored in the URL and it has higher priority than the default value.

**Comment by jinghua-qa** [comment]
@michael-s-molina is this new behavior for native filter? I checked on preset prod and the old behavior is : when user set default value, after edit the filter value in dashboard, the default value will come back when page refreshed. Also I think my understanding for default value is, when the filter value is set at empty(like user click clear all, or delete), page refresh will show default value. Now if the value is edited to empty, it just stay empty.

**Comment by villebro** [comment]
This behavior is consistent with how e.g. Explore works. But I wonder if it's expected behavior. The upside of this is it's possible to make a filter selection and directly share the URL with someone, but the downside is to reset to defaults you have to go to the dashboard page and open it again (or manually remove the query params in the URL).

**Comment by michael-s-molina** [comment]
> @michael-s-molina is this new behavior for native filter? I checked on preset prod and the old behavior is : when user set default value, after edit the filter value in dashboard, the default value will come back when page refreshed. 

Yes. It's a new behavior. Previously, we didn't support saving the applied filters in the URL.

> Also I think my understanding for default value is, when the filter value is set at empty(like user click clear all, or delete), page refresh will show default value. Now if the value is edited to empty, it just stay empty.

It's a trade-off decision. Personally, I prefer always to preserve the user selection. Imagine the scenario where a user is exploring a dashboard with a bunch o filters, then at some point the user changes some values, clears some filters, and finds the exact information he was looking for. Then the user copies the URL and sends it to someone else. The user is expecting that the person will see the same data. If we change the current implementation, we can break this workflow.

I also think is very easy for the user to select the default value again, so for me preserving the user selections is more important. What do you all think?

**Comment by junlincc** [comment]
User should not be able to change a default value from the native filter bar, they should only change it from the modal. i think?

**Comment by jinghua-qa** [comment]
any update for this issue? Do we have conclusion for the correct behavior for default native filter?
